lib/CleantalkSP/SpbctWP/AdjustToEnvironmentModule/AdjustToEnvironmentSettings.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+namespace CleantalkSP\SpbctWP\AdjustToEnvironmentModule;
+
+class AdjustToEnvironmentSettings
+{
+    public static function render()
+    {
+        $view = '';
+        $view_buttons = '';
+
+        $view .= '<div class="spbc_wrapper_field" id="action-shuffle-salts-wrapper">';
+        $view .= __('We changed some settings of third party plugins to make compatibility. ', 'security-malware-firewall');
+        $view .= __('You are free to reverse it, but it may lead to incorrect operation of the software.', 'security-malware-firewall');
+
+        $info = AdjustToEnvironmentHandler::getInfoWhatWeChanged();
+
+        foreach ( array_values(AdjustToEnvironmentHandler::SET_OF_ADJUST) as $adjust_class ) {
+            $key_of_adjust = array_search($adjust_class, AdjustToEnvironmentHandler::SET_OF_ADJUST);
+            if (isset($info[$adjust_class]['changed']) && $info[$adjust_class]['changed']) {
+                $view_buttons .= self::renderReverseBtn($key_of_adjust);
+                continue;
+            }
+
+            $adjust = new $adjust_class([]);
+            // if site admin request to reverse adjust once before, we suggest to make the adjust using the button
+            if (isset($info[$adjust_class]['changed']) && !$info[$adjust_class]['changed'] && $adjust->isNeedToAdjustOne()) {
+                $view_buttons .= self::renderChangeBtn($key_of_adjust);
+            }
+        }
+
+        $view .= $view_buttons;
+        $view .= '</div>';
+
+        if ($view_buttons === '') {
+            $view = '';
+        }
+
+        return $view;
+    }
+
+    private static function renderChangeBtn($key)
+    {
+        $btn = self::btnLayout();
+
+        $title = '';
+        $description = '';
+
+        switch ($key) {
+            case 'w3tc':
+                $title = 'W3 Total Cache';
+                $description = __('Next button will disable "Cache URIs with query string variables" option.', 'security-malware-firewall');
+                break;
+        }
+
+        $replaces = [
+            '{TITLE}' => __("Make adjust for $title", 'security-malware-firewall'),
```

Comment:
It is not allowed to pass variables into the translation string. Pass it via `sprintf()`

Reply:
Fixed, thanks.